bin/auto-update/index.ts

```diff
@@ -0,0 +1,74 @@
+import { join, resolve } from 'path';
+import { SH } from './sh';
+import { Git } from './git';
+import { Helpers } from './helpers';
+
+if (!process.env.GH_AUTH_TOKEN) {
+  throw new Error('Please, set env var: GH_AUTH_TOKEN');
+}
+
+export interface Settings {
+  dtForkPath: string; // full path to the local DT fork folder
+  typesDirName: string; // directory name in DT
+  typesBranchName: string; // branch name where generated types are in the generator repo
+  user: string; // user who submits PRs to DT
+  auth: string; // GH token with public_repo access
+  dtRepoOwner: string; // DefinitelyTyped repo owner name only
+  dtRepoName: string; // DefinitelyTyped repo name only
+  thisRepo: string; // repo form where API calls to GH will be made
+}
+
+const settings: Settings = {
+  dtForkPath: join(resolve(__dirname, '../..'), 'dt-fork'), // TODO: maybe use require('os').tmpdir() or something like that
```

Comment:
Thanks for reminding about `.tmp` :)
The idea was that at this point `types` branch is already tested and nothing can go wrong, but since we're running `npm test` from DT fork, their tests might differ from ours.
So yes, we will still need to have that `.tmp` and artifacts publishing.
However, I don't think that we should publish `dt-fork` as an artifact, it's huge.
But in order to trace bugs with generated types, we might want to log the commit hash of `types` branch that was used for this run, and upload whole `.tmp` folder as an artifact. I think it should be simple and sufficient. I still have to add action for this. Did I get your question right?

Reply:
> Did I get your question right?

Actually, I was just referring to temp directory path. It might nice to keep both of these temporary things (DT fork and Google API JSON) in the same temp directory for consistency.

But you're making a fair point: If we are running DT's tests, ideally we would understand why it failed.

> and upload whole `.tmp` folder as an artifact

The commit hash of the `types` branch could be useful, but we're not actually generating the types in this flow, so I'm not sure what you mean by this.

My suggestion would be to _not_ run the DT tests for this initial rollout, and we can see how often they fail. I would hope our test implementation is basically doing the same thing. I understand if you want to keep them in there and am happy to chat more about improving the transparency of failures.

Reply:
> Actually, I was just referring to temp directory path. It might nice to keep both of these temporary things (DT fork and Google API JSON) in the same temp directory for consistency.

Oh, got it. I agree. We might want to have this structure:
```
tmp
  - api-discovery-json
  - dt-fork
```
I will create a task to find the best way of getting/creating `tmp` folder (`require('os').tmpdir()` or something like this).

> The commit hash of the types branch could be useful, but we're not actually generating the types in this flow, so I'm not sure what you mean by this.
Right, we don't need to upload JSON API if DT tests failed.
If DT tests failed - that means that our `dtslint` didn't catch the issue when generating `types` branch. So, I think that we only need `types` commit hash to investigate this. Probably, also DT commit hash. Oh, and `package-lock.json` from `dt-fork` because they have `"dtslint": "latest",` in their `package.json` and no lock-file. It might be important for investigation.

> My suggestion would be to not run the DT tests for this initial rollout, and we can see how often they fail. I would hope our test implementation is basically doing the same thing. I understand if you want to keep them in there and am happy to chat more about improving the transparency of failures.

Well, then we'll be violating DT PR template :) Also, I'm not sure that our `dtslint` does everything that [microsoft/types-publisher](https://github.com/microsoft/types-publisher) does. For example, it also checks for conflicts with other types, etc. Hence, I think we should keep it for now.

Reply:
Sounds good. Thanks for creating https://github.com/Maxim-Mazurok/google-api-typings-generator/issues/113!